### PR TITLE
Various bug fixes (real root extraction + changes of variables)

### PR DESCRIPTION
**Pull request by mohabsafey**
* fixes bug occurring when changing variables order
* fixes bug occurring after some linear changes of variables
* fixes bug in the real root extraction

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -33,7 +33,9 @@ checkdiff               = test/diff/diff_cp_d_3_n_4_p_2.sh \
 			  test/diff/diff_radical_shape-31.sh \
 			  test/diff/diff_radical_shape-qq.sh \
 			  test/diff/diff_reals_dim0.sh \
-			  test/diff/diff_reals_dim0-swapvar.sh \
+				test/diff/diff_reals_extract.sh \
+				test/diff/diff_reals_dim0-swapvar.sh \
+				test/diff/diff_reals_dim0-chgvar.sh \
```
Comment (ederc) [inline Makefile.am:38]: Please fix indentation.

File: src/msolve/msolve.c [hunk 2]
```diff
@@ -3672,9 +3671,10 @@ int newvalue_denom(mpz_t *denom, long deg, mpz_t r, long k,
                    long corr, mpz_t c){
 
   mpz_add_ui(c, r, 1);
+  /*boo = 1 if sgn(den_do) != sgn(den_up) else it is 0*/
   int boo = mpz_poly_eval_interval(denom, deg, k,
-                               r, c,
-                               tmp, den_do, den_up);
+                                   r, c,
+                                   tmp, den_do, den_up);
```
Comment (ederc) [inline src/msolve/msolve.c:3677]: Please fix indentation.

**Comment by ederc** [comment]
Please rebase on current master.

File: Makefile.am (hunk 1 above)
Comment (mohabsafey) [reply Makefile.am]: Should be done now

File: src/msolve/msolve.c (hunk 2 above)
Comment (mohabsafey) [reply src/msolve/msolve.c]: Should be done now

**Pull request merged**